CLAUDE.md

```
@@ -180,6 +180,70 @@ const data: MyType = JSON.parse(raw);
+
+- **MUST** put one concept per file. A file named `foo.ts` should be readable in one sitting and describable in one sentence.
+- **MUST** name files for the concept they contain, not generic buckets like `utils.ts` / `helpers.ts` / `common.ts`. Generic filenames attract unrelated additions from every PR and become conflict magnets.
+- **MUST** put shared helpers in their own named files under the appropriate directory (`server/utils/fs.ts`, `server/utils/errors.ts`) rather than dumping them into the file that first needs them. Callers import from the helper's file, not the file that happens to re-export it.
+- **SHOULD** split a file once it grows past ~500 lines OR once it has two concepts that PRs keep touching independently.
+- **MUST NOT** create re-export barrel files (`index.ts` that re-exports from siblings) unless there is a specific reason. They create phantom merge conflicts and hide the real dependency graph.
+
+### Directories: the name is the contract
+
+Looking at a directory name should be enough to predict what's inside. The existing top-level structure is the reference:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)